L1Trigger/L1TCommon/src/XmlConfigParser.cc

```diff
@@ -0,0 +1,647 @@
+  kAttrType(nullptr),
+  kAttrDelim(nullptr),
+  kAttrModule(nullptr),
+  kTypeTable("table")
+{
+  XMLPlatformUtils::Initialize();
+
+  kTagHw          = XMLString::transcode("system");
+  kTagAlgo        = XMLString::transcode("algo");
+  kTagRunSettings = XMLString::transcode("run-settings");
+  kTagDb          = XMLString::transcode("db");
+  kTagKey         = XMLString::transcode("key");
+  kTagLoad        = XMLString::transcode("load");
+  kTagContext     = XMLString::transcode("context");
+  kTagParam       = XMLString::transcode("param");
+  kTagMask        = XMLString::transcode("mask");
+  kTagDisable     = XMLString::transcode("disable");
+  kTagExclBoards  = XMLString::transcode("excluded-boards");
+  kTagExclude     = XMLString::transcode("exclude");
+  kTagColumns     = XMLString::transcode("columns");
+  kTagTypes       = XMLString::transcode("types");
+  kTagRow         = XMLString::transcode("row");
+  kTagProcessor   = XMLString::transcode("processor");
+  kTagRole        = XMLString::transcode("role");
+  kTagCrate       = XMLString::transcode("crate");
+  kTagSlot        = XMLString::transcode("slot");
+  kTagDaqTtc      = XMLString::transcode("daqttc-mgr");
+  kAttrId         = XMLString::transcode("id");
+  kAttrType       = XMLString::transcode("type");
+  kAttrDelim      = XMLString::transcode("delimiter");
+  kAttrModule     = XMLString::transcode("module");
+ 
+  parser_ = nullptr; 
+  doc_ = doc;
+}
+
+
+XmlConfigParser::~XmlConfigParser()
+{
+  delete parser_;
+  XMLPlatformUtils::Terminate();
+}
+
+
+void XmlConfigParser::readDOMFromString(const std::string& str, DOMDocument*& doc)
+{
+  MemBufInputSource xmlstr_buf((const XMLByte*)(str.c_str()), str.size(), "xmlstrbuf");
+  parser_->parse(xmlstr_buf);
+  doc = parser_->getDocument();
+  assert(doc);
+}
+
+
+void XmlConfigParser::readDOMFromString(const std::string& str)
+{
+  MemBufInputSource xmlstr_buf((const XMLByte*)(str.c_str()), str.size(), "xmlstrbuf");
+  parser_->parse(xmlstr_buf);
+  doc_ = parser_->getDocument();
+  assert(doc_);
+}
+
+
+void XmlConfigParser::readDOMFromFile(const std::string& fName, DOMDocument*& doc)
+{
+  parser_->parse(fName.c_str()); 
+  doc = parser_->getDocument();
+
+  if (! doc) {
+    edm::LogError("XmlConfigParser") << "Could not parse file " << fName << "\n";
+  }
+
+  assert(doc);
+}
+
+
+void XmlConfigParser::readDOMFromFile(const std::string& fName)
+{
+  parser_->parse(fName.c_str()); 
+  doc_ = parser_->getDocument();
+
+  if (! doc_) {
+    edm::LogError("XmlConfigParser") << "Could not parse file " << fName << "\n";
+  }
+
+  assert(doc_);
+}
+
+
+void XmlConfigParser::readRootElement(TriggerSystem& aTriggerSystem, const std::string& sysId)
+{
+  DOMElement* rootElement = doc_->getDocumentElement();
+  if (rootElement) {
+    if (rootElement->getNodeType() == DOMNode::ELEMENT_NODE) {
+      readElement(rootElement, aTriggerSystem, sysId);
+    }
+  } else {
+    std::cout << "No xml root element found" << std::endl;
```

Comment:
That'll bring a dependency on the Logging library of CMSSW. I don't know if Thomas Reis (the author of XmlConfigReader, that this class is derived from) had ever intended to use it in a standalone mode. I'll better try to clean all the std::cout's instead.